### PR TITLE
make a minimal server-side retrieval example

### PR DESCRIPTION
**Pull request by zachcp**
This adds a minimal example wherein a GET call is used to retrieve datoms from the server and add them to the re-posh/datascript DB.

Not sure if you want this but this is related to issue #17.

### CODE REVIEW

**Comment by denistakeda** [comment]
Hi @zachcp . Sorry for reviewing it so late. Looks like I can't query `/upload` endpoint. Have you tested it?

**Comment by zachcp** [comment]
This works on my end. I did not test via REST/curl. But if you click the button you inject the extra entries. 

```
cd examples/todomvc/
lein run

# navigate to http://127.0.0.1:5000/ and click
```

File: examples/todomvc/src/cljs/todomvc/events.cljs [hunk 1]
```diff
@@ -21,8 +22,26 @@
...
+(re-frame/reg-event-fx
+  :retrieve
+  [(re-frame/inject-cofx :ds)] ;; inject coeffect
+  (fn [{:keys [ds]} [_ signal params]] ;; ds here is the DataScript database
+    (GET
+      "/upload"
+      :format :edn
+      :handler #(re-frame/dispatch  [:add-entries  %1])
+      :error-handler #(js/alert (str "Retrieval of entries failed!")))
+    ds))
```
Comment (denistakeda) [inline examples/todomvc/src/cljs/todomvc/events.cljs:41]: In this particular case, you need nothing from outside world in this event. So, no need for coeffect injecting and function signature can be much shorter. Here the adjusted variant:
```clojure
(re-frame/reg-event-fx
  :retrieve
  (fn [] 
    (GET
      "/upload"
      :format :edn
      :handler #(re-frame/dispatch  [:add-entries  %1])
      :error-handler #(js/alert (str "Retrieval of entries failed!")))
    {}))
```
With this everything works just fine. I would also recomment using [effect](https://github.com/Day8/re-frame/blob/master/docs/Effects.md) for sending events, not the pure event (as far as you are not going to do anything with database)

**Comment by denistakeda** [comment]
@zachcp  I've seen this PR as having learning purpose.  I wanna keep this `todo-mvc` example as simple as possible. It's better to create the new one with backend and everything around. But if you achieve your goal maybe I can close this one? What do you think?

**Comment by zachcp** [comment]
Indeed. I would agree with that. Thanks for your help. Zcp

**Pull request closed without merging**